erpnext/stock/dashboard/item_dashboard.js

```diff
@@ -75,13 +75,22 @@ erpnext.stock.ItemDashboard = Class.extend({
 			this.content.find('.more').addClass('hidden');
 		}
 
-        // If not any stock in any warehouses provide a message to end user
+		// If not any stock in any warehouses provide a message to end user
 		if (context.data.length > 0) {
 			$(frappe.render_template('item_dashboard_list', context)).appendTo(this.result);
 		} else {
 			var message = __(" Currently no stock available in any warehouse")
 			$("<span class='text-muted small'>"+message+"</span>").appendTo(this.result);
 		}
+
+		if (frappe.user_roles.indexOf("Stock Manager") > -1 || frappe.user_roles.indexOf("Stock User") > -1){
```

Comment:
Instead of checking hardcode roles, check user has write permissions for Stock Entry doctype
frappe.boot.user.can_write.indexOf("Stock Entry")